ddl/ddl_tiflash_api.go

```diff
@@ -406,6 +406,8 @@ func pollAvailableTableProgress(schemas infoschema.InfoSchema, _ sessionctx.Cont
 				// https://github.com/pingcap/tidb/issues/39949
 				panic(err)
 			}
+			pollTiFlashContext.UpdatingProgressTables.Remove(element)
```

Comment:
Why does removing elements solve this problem?

Reply:
The outside loop will iterate for `min(pollMaxCount, UpdatingProgressTables.Len())`. If we `continue` here, the loop will carry on with `pollMaxCount -= 1`. However the iterator of `UpdatingProgressTables` will not be advanced. That means the same element will be handled for `pollMaxCount` times.

We have observed many requests of the same table to PD in less than 1 sec.

Reply:
Do we have test for this logic? If it hard to add UT, please add a manual test description.
Rest LGTM.